packages/hardhat-chai-matchers/test/reverted/revertedWithCustomError.ts

```diff
@@ -456,6 +456,18 @@ describe("INTEGRATION: Reverted with custom error", function () {
           "sender doesn't have enough funds to send tx"
         );
       });
+
+      it("extra arguments", async function () {
+        expect(() => {
+          const assertExtraArguments = expect(
+            matchers.revertWithSomeCustomError()
+          ).to.be.revertedWithCustomError as (...args: any[]) => any;
```

Comment:
Ah, is it because it won't compile otherwise? I think the right thing to do is to use `@ts-expect-error`, see for example this other test:

https://github.com/NomicFoundation/hardhat/blob/d9d5d193a9076920ba0bd899d515b2cac364a685/packages/hardhat-chai-matchers/test/reverted/revertedWith.ts#L200-L201